plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -50,16 +51,62 @@
 _default_logger = logging.getLogger(__name__)
 
 _ETHEREUM = "ethereum"
-GAS_ID = "gwei"
 TESTNET_NAME = "ganache"
 DEFAULT_ADDRESS = "http://127.0.0.1:8545"
 DEFAULT_CHAIN_ID = 1337
-DEFAULT_GAS_PRICE = "50"
 DEFAULT_CURRENCY_DENOM = "wei"
 _ABI = "abi"
 _BYTECODE = "bytecode"
 
 
+def get_gas_price_strategy(
+    gas_price_strategy: Optional[str] = None, api_key: Optional[str] = None
+) -> Callable:
+    """Get the gas price strategy."""
+    supported_gas_price_modes = ["safeLow", "average", "fast", "fastest"]
+    if gas_price_strategy is None:
+        _default_logger.debug(
+            "Gas price strategy not provided. Falling back to `rpc_gas_price_strategy`."
+        )
+        return rpc_gas_price_strategy
+
+    if gas_price_strategy not in supported_gas_price_modes:
+        _default_logger.debug(
+            "Gas price strategy `{gas_price_strategy}` not in list of supported modes: {supported_gas_price_modes}. Falling back to `rpc_gas_price_strategy`."
+        )
+        return rpc_gas_price_strategy
+
+    if api_key is None:
+        _default_logger.debug(
+            "No ethgasstation api key provided. Falling back to `rpc_gas_price_strategy`."
+        )
+        return rpc_gas_price_strategy
+
+    def gas_station_gas_price_strategy(web3: Web3, transaction_params: TxParams) -> int:
+        """
+        Get gas price from ethgasstation api.
+
+        Visit `https://docs.ethgasstation.info/gas-price` for documentation.
+        """
+        response = requests.get(
+            "https://ethgasstation.info/api/ethgasAPI.json?api-key={api_key}"
+        )
+        if response.status_code != 200:
+            raise ValueError(
+                "Gas station API response: {response.status_code}, {response.text}"
+            )
+        response_dict = json.loads(response.text)
+        _default_logger.debug("Gas station API response: {}".format(response_dict))
+        result = response_dict.get(gas_price_strategy, None)
+        if type(result) not in [int, float]:
+            raise ValueError("Invalid return value for `{gas_price_strategy}`!")
+        gwei_result = result / 10
```

Comment:
This seems unexpected - an explanation could help here

Reply:
+1

I believe it is due to ETH Gas Station APIs:
https://docs.ethgasstation.info/gas-price

yet, a comment would be great for future reference.